app/auth/routes.py

```diff
@@ -83,3 +83,20 @@ def reset_password(token):
         flash('Your password has been reset.')
         return redirect(url_for('auth.login'))
     return render_template('auth/reset_password.html', title='Reset Password', form=form)
+
+@bp.route('/reset_password_preferences', methods=['GET', 'POST'])
+def reset_password_preferences():
+    user = current_user
+    if not user:
+        return redirect(url_for('main.index'))
```

Comment:
You need to add the @login_required decorator to the route. It goes between the route decorator and the class definition like this. 
![image](https://user-images.githubusercontent.com/62105185/202522273-35728a88-92d7-4ec7-9e94-cab84cc2f9d8.png)

With it added, you can get rid of the: if not user: and redirect return line.